src/main/scala/com/ing/wbaa/rokku/sts/service/ExpiredTokenCleaner.scala

```diff
@@ -0,0 +1,47 @@
+package com.ing.wbaa.rokku.sts.service
+
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+import akka.actor.{ Actor, ActorSystem, Props, Timers }
+import com.ing.wbaa.rokku.sts.data.aws.AwsSessionTokenExpiration
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.concurrent.duration._
+import scala.concurrent.{ ExecutionContext, Future }
+
+trait ExpiredTokenCleaner {
+
+  protected[this] implicit def system: ActorSystem
+
+  protected[this] def cleanExpiredTokens(expirationDate: AwsSessionTokenExpiration): Future[Int]
+
+  system.actorOf(Props(new ExpiredTokenCleanActor(cleanExpiredTokens)), "expiredTokenCleanActor")
+}
+
+object ExpiredTokenCleanActor {
+
+  private case object CleanerKey
+
+  private case object CleanToken
+
+}
+
+class ExpiredTokenCleanActor(cleanExpiredTokens: AwsSessionTokenExpiration => Future[Int]) extends Actor with Timers with LazyLogging {
```

Comment:
I think using an actor for this is an overkill (we don't even send messages to it). Instead, I would use `system.scheduler.schedule(Duration, Duration, Runnable)`.

Reply:
you do not like actors :P